scripts/mist-cleanup.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+script=$(basename "$0")
+pidfile="/var/run/${script}"
+
+# lock so that multiple instances of this script cannot run in parallel
+exec 200>"$pidfile"
+flock -n 200 || exit 1
+pid=$$
+echo $pid 1>&200
+
+log () {
+  echo $$ - $(date) - "mist-cleanup" - $1 2>&1
+}
+
+containsElement () {
+  local e match="$1"
+  shift
+  for e; do [[ "$e" == "$match" ]] && return 0; done
+  return 1
+}
+
+SEMS=$(ls -tr /dev/shm/sem.MstTRKS* /dev/shm/sem.MstUser* /dev/shm/sem.MstInpt* /dev/shm/sem.MstPull* 2> /dev/null)
+PULS=$(ls -tr /dev/shm/sem.MstPull* 2> /dev/null)
+#sem.MstPull_golive+Preview
+PAGS=$(find /dev/shm/ -name "MstData*" -o -name "MstMeta*" -o -name "MstTrak*" 2> /dev/null)
+STAT=$(ls -tr /dev/shm/MstSTATEgolive* 2> /dev/null)
+ACTS=$(ps -C MistInBuffer -o command= | awk ' { print $3 } ' | sort | uniq)
+DTPL=$(ps -C MistInDTSC -o command= | awk ' { print $3 } ' | sort | uniq)
+DATA_DEL=0
+
+log "Starting..."
+
+#Check for input semaphores without data
+for S in $SEMS ; do
+  STRM=${S:20}
+  if containsElement "$STRM" "$ACTS" ; then
+    log "Yes: ${STRM} (${S})"
+  else
+    log "SEM_INPUT: ${S}"
+    rm "${S}"
+  fi
+done
+
+#Check for pull semaphores without data
+for S in $PULS ; do
+  STRM=${S:21}
+  if containsElement "$STRM" "$DTPL" ; then
+    log "Yes: ${STRM} (${S})"
+  else
+    log "SEM_PULL: ${S}"
+    rm "${S}"
+  fi
+done
+
+#Check for data pages without buffer
+for P in $PAGS; do
+  FILE=${P%@*} #Remove everything after the '@'
+  STRM=${FILE:16} #Strip the beginning
+  if containsElement "$STRM" "$ACTS" ; then
+    log "Yes: ${STRM} (${P})"
+  else
+    if (( $# != 0 )); then
+      log "DATA: ${P}"
+    fi
+    rm "${P}"
+    DATA_DEL=$(( DATA_DEL + 1 ))
+  fi
+done
+
+#Check for state pages without buffer
+for P in $STAT ; do
+  STRM=${P:17} #Strip the beginning
+  if containsElement "$STRM" "$ACTS" ; then
+    log "Yes: ${STRM} (${P})"
+  else
+    log "STATE: ${P}"
+    rm "${P}"
+  fi
+done
+
+log "Done. Deleted ${DATA_DEL} data pages"
```

Comment:
Is it worth calculating and displaying number of deleted semaphores and state pages?